packages/ckeditor5-utils/src/collection.ts

```diff
@@ -91,7 +101,7 @@ export default class Collection {
 		 * @private
 		 * @member {String}
 		 */
-		this._idProperty = options.idProperty || 'id';
+		this._idProperty = options.idProperty || 'id' as I;
```

Comment:
The cast wouldn't be necessary if we make options mandatory (so we can check `idProperty` against `I` type parameter). I think it would be possible to only do it for non-default `idProperty` using conditional types. That would reduce readability